wodles/gcloud/gcloud.py

```diff
@@ -20,18 +26,32 @@
     credentials_file = arguments.credentials_file
     max_messages = arguments.max_messages
     log_level = arguments.log_level
+    n_threads = arguments.n_threads
 
     # get logger
     logger = tools.get_stdout_logger(tools.logger_name, log_level)
 
-    # get Google Cloud client
-    client = WazuhGCloudSubscriber(credentials_file, project, subscription_id)
+    if n_threads > max_threads:
+        n_threads = max_threads
+        logger.warning(f'Reached maximum number of threads. Truncating to {max_threads}')
 
-    # check permissions about subscription
-    client.check_permissions()
+    logger.debug(f"Setting {n_threads} thread{'s' if n_threads > 1 else ''} to pull {max_messages}" 
+                 f" message{'s' if max_messages > 1 else ''} each")
 
     # process messages
-    num_processed_messages = client.process_messages(max_messages)
+    with ThreadPoolExecutor() as executor:
+        futures = []
+
+        # check permissions
+        subscriber_client = WazuhGCloudSubscriber(credentials_file, project, subscription_id)
+        subscriber_client.check_permissions()
+        futures.append(executor.submit(subscriber_client.process_messages, max_messages // n_threads))
+
+        for _ in range(n_threads - 1):
+            client = WazuhGCloudSubscriber(credentials_file, project, subscription_id)
+            futures.append(executor.submit(client.process_messages, max_messages // n_threads))
```

Comment:
I pushed a commit to the branch with some changes to fix that bug and speed up the program a little bit.
https://github.com/wazuh/wazuh/blob/8d88755f04cfb0b3349972b5a9a66e3eac5f2b53/wodles/gcloud/gcloud.py#L46-L55

- Now if there are 3 threads and `max_messages` is 100, the number of messages that the program will process won't be 99 but 100.
- If the number of threads is greater than the number of messages, all the messages will be processed by one thread.
- The calculations of the number of messages per thread are now out of the loop since it was not necessary to compute that for every iteration, but just once.